src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java

```diff
@@ -50,117 +54,132 @@ public String getUrlForPage(String wikiName, String pageName) {
+  }
+
+  public String getUrlForWiki(String wikiName, String language) {
+    return getUrlForWiki(wikiName, language, false, envType);
+  }
+
+  public String getUrlForWiki(String wikiName, String language, boolean addWWW, EnvType envType) {
     final String wikiaName = getWikiaGlobalName(wikiName);
 
-    String www = "";
-    if (addWWW) {
-      www = "www.";
+    HttpUrl.Builder urlBuilder = new HttpUrl.Builder();
+
+    String www = addWWW ? "www." : "";
+    String host = getFormattedWikiHost(www, wikiaName, envType, language);
+
+    if (!("en").equals(language) && forceLanguageInPath) {
+        urlBuilder.addEncodedPathSegments(language);
     }
+    return urlBuilder.scheme(getUrlProtocol()).host(host).build().toString();
+  }
+
+  private String getFormattedWikiHost(String www, String wikiaName, EnvType envType, String language) {
+    if (!forceLanguageInPath)
+      wikiaName = String.join(".", language, wikiaName);
+    return getFormattedWikiHost(www, wikiaName, envType);
+  }
 
+  private String getFormattedWikiHost(String www, String wikiaName, EnvType envType) {
     switch (envType) {
       case DEV: {
         String devBoxOwner = this.env.split("-")[1];
-        return String.format(getUrlProtocol() + DEV_URL_FORMAT, www, wikiaName, devBoxOwner, envType.getWikiaDomain());
+        return String.join(".", www + wikiaName, devBoxOwner, envType.getWikiaDomain());
       }
       case PROD: {
-        return String.format(getUrlProtocol() + PROD_URL_FORMAT, www, wikiaName, envType.getWikiaDomain());
+        return String.join(".", www + wikiaName, envType.getWikiaDomain());
       }
       case STAGING: {
-        return String.format(getUrlProtocol() + PROD_URL_FORMAT, www, wikiaName, envType.getWikiaDomain());
+        return String.join(".", www + wikiaName, envType.getWikiaDomain());
       }
       case SANDBOX: {
-        if (newStagingUrlFormat) {
-          return String.format(getUrlProtocol() + SANDBOX_NEW_URL_FORMAT, www, wikiaName, this.env, envType.getWikiaDomain());
-        } else {
-          return String.format(getUrlProtocol() + SANDBOX_URL_FORMAT, this.env, www, wikiaName, envType.getWikiaDomain());
-        }
+        return String.join(".", www + wikiaName, this.env, envType.getWikiaDomain());
       }
       default:
-        return "";
+        throw new WebDriverException("Unknown environment type");
     }
   }
 
-  public String getUrlProtocol() {
-    if (this.forceHttps) {
-      return HTTPS_PREFIX;
-    }
-    return HTTP_PREFIX;
-  }
-
-  public String getWikiGlobalURL() {
-    EnvType env = Configuration.getEnvType(this.env);
-
+  private String getFormattedHostForGlobalUrl(EnvType env) {
     switch (env) {
       case DEV: {
         String devBoxOwner = this.env.split("-")[1];
-        return String.format(getUrlProtocol() + "www.%s.%s", devBoxOwner, env.getWikiaDomain());
+        return String.join(".", "www", devBoxOwner, envType.getWikiaDomain());
       }
       case SANDBOX: {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested code block into a method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1199)